pkg/cvo/availableupdates.go

```diff
@@ -95,12 +108,24 @@ func (u *availableUpdates) NeedsUpdate(original *configv1.ClusterVersion) *confi
 
 // setAvailableUpdates updates the currently calculated version of updates.
 func (optr *Operator) setAvailableUpdates(u *availableUpdates) {
+	success := false
 	if u != nil {
-		u.At = time.Now()
+		u.LastAttempt = time.Now()
+		if u.Condition.Type == configv1.RetrievedUpdates {
+			success = u.Condition.Status == configv1.ConditionTrue
+		} else {
+			klog.Warningf("Unrecognized condition %s=%s (%s: %s): cannot judge update retrieval success", u.Condition.Type, u.Condition.Status, u.Condition.Reason, u.Condition.Message)
+		}
 	}
 
 	optr.statusLock.Lock()
 	defer optr.statusLock.Unlock()
+	if u != nil && (optr.availableUpdates == nil ||
+		optr.availableUpdates.Upstream != u.Upstream ||
+		optr.availableUpdates.Channel != u.Channel ||
+		success) {
+		u.LastSyncOrConfigChange = u.LastAttempt
+	}
```

Comment:
LastSyncOrConfigChange is getting clobbered after its initially set. Need below change or similar:

`        if u != nil {
                if optr.availableUpdates == nil ||
                        optr.availableUpdates.Upstream != u.Upstream ||
                        optr.availableUpdates.Channel != u.Channel ||
                        success {
                        u.LastSyncOrConfigChange = u.LastAttempt
                } else if optr.availableUpdates != nil {
                        u.LastSyncOrConfigChange = optr.availableUpdates.LastSyncOrConfigChange
                }
        }`

Reply:
Good catch; should be fixed with 326ccf5 -> 3812e5f.